db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/QualifierPhase.java

```diff
@@ -38,8 +38,9 @@
  * BASE - > Qualifier is going against a key value in a way where splice can perform partitioning.
  * FILTER_BASE -> Qualifier for base table but does not restrict I/O from base scan.  This is evaluated prior to a possible lookup.
  * FILTER_PROJECTION -> Qualifer applied after possible lookup in the projection node on top of the base conglomerate scan.
+ * JOIN -> Join predicate used to adjust the join selectivity.
  *
  */
 public enum QualifierPhase {
-    BASE,FILTER_BASE,FILTER_PROJECTION
+    BASE,FILTER_BASE,FILTER_PROJECTION,JOIN
```

Comment:
One thing that could lead to some confusion later is that the JOIN phase newly added here is not used in `addPredicate()` in `ScanCostFunction`, but only used in `estimateJoinSelectivity()` in `SelectivityUtil`. For this reason, I wonder if it's necessary to have this phase. In `estimateJoinSelectivity()`, there is only one phase involved, so practically, phase doesn't really matter.

Reply:
Good point.  I will remove it and just pass FILTER_BASE as the parameter value.